### DISCUSSION TITLE
Practicum Submission | Arseno Feri Alzahabi | arse | October

### DISCUSSION THREAD

**Discussion by arsenoferi** [onlyphantom/llm-python#42] [Practicum: Generative AI]
Arseno Feri Alzahabi | arse

Hi Team,

Attached below is the link of my ipynb file in github
Github Link  : [(https://github.com/arsenoferi/Supertype_Gen_Ai_finance_challange/blob/main/Generative_AI_and_Chatbot_challenge.ipynb)

Best regards,
Arseno Feri Alzahabi

**Comment by onlyphantom** [comment]
a couple of feedback! 

1. there are ways to write more specific assertion / tests for `test_output_parser_symbol_invalid` and `test_output_parser_mcap_invalid` so you check specifically for cases where symbol is not 4 letters long (BBCA, BBRI, etc) or when the extraction takes you to a value of <0 on market cap

2. you shouldn't have to prepend the content of `out_agent[-1].content` in `query_app(f"{out_agent[-1].content}. get overview 4th company")` because the point of the exercise is to test that your ai has memory and remember its context from the conversation. So pre-pending or appending past history kinda help the ai "cheat" and didnt give a clear indication of whether it actually did have memory.

Overall, solid effort. Grading it as a pass!
